### ISSUE TITLE
templates

### ISSUE THREAD

**Issue by joshinils** [philsupertramp/wiki#10]
# templates
quasi funktionen mit argumenten und einem (text) resultat (das interpretiert und dargestellt werden kann), namespace `template`?

syntax vorschlag: `{{templatename|arg1}}` optionale argumente, default werte möglich bei der definition, mehrere argumente möglich bei einer template-definition
* zum einbinden anderer seiten und oder deren inhalt wie zb per `#include`
* zum programmieren von inhalten anhand von parametern für zb. icons,
* generierung von dingen nach bausätzen
* möglichkeit code-blöcke formattiert darzustellen, also per aufruf eines templates, format-typ und inhalt als argumente, resultat ist eine box mit code. zb mit zeilennummern, farbig. sowas einstellbar

PZ: Ok, könnten wir uns evtl darauf einigen die Dinger "Templatetags" oder sowas zu nennen?
Templates sind in meiner Welt vorgefertigte Layouts in die ich nur noch variablen reinkloppe, die dann richtig dargestellt werden.
Für mich klingt deine Version eher nach funktionen, die du in deiner seite rufen willst, die unterschiedliche Dinge können. Z.b {{today}} würde vlt 10.02.2021 ausgeben, oder {{Startseite}} einen Link zur Startseite.
Stellst du dir dazu dann auch einen Skope vor? Ansonsten könnte man ja code blocks nicht wirklich formatieren, müsste dann ja sowas sein wie
````
{{cpp_code_burning|on}}
```
int main(){
	return 0;
}
```
{{cpp_code_burning|off}}
````
oder sowas.

**Comment by joshinils** [comment]
ja, eigentlich sind es funktionen, die "magisch" sachen können, aber halt eine eigene seite im template namensraum haben mit deren definition (oder halt nicht, wenn sie sowas wie `{{today}}` sind) 

ich mag die schreibweise oben mit dem on und off nicht, da ich davon ausgehe, dass das ebnen nicht so funkltioniert wie in latex, dass der scope einer solchen funktion die geschweiften klammern sind

**Comment by joshinils** [comment]
frage wäre wie man die funktionalität dann nennen soll, wenn nicht templates.
* funktionen - kommt mir zu generisch vor
* template - überlagerung mit einem anderen begriff bei dir
* text-modul - kommt drauf an was das hinterher können soll, die bezeichnung soll das ja nicht so sonderlich einschränken.

**Comment by philsupertramp** [comment]
> ja, eigentlich sind es funktionen, die "magisch" sachen können, aber halt eine eigene seite im template namensraum haben mit deren definition (oder halt nicht, wenn sie sowas wie `{{today}}` sind) 
> 
> ich mag die schreibweise oben mit dem on und off nicht, da ich davon ausgehe, dass das ebnen nicht so funkltioniert wie in latex, dass der scope einer solchen funktion die geschweiften klammern sind

Dann kann ich mir aber noch nicht vorstellen wie ein skope entsteht, oder packst du dann alles in den Aufruf des TeMpLaTeS? 😄

**Comment by joshinils** [comment]
> > ja, eigentlich sind es funktionen, die "magisch" sachen können, aber halt eine eigene seite im template namensraum haben mit deren definition (oder halt nicht, wenn sie sowas wie `{{today}}` sind)
> > ich mag die schreibweise oben mit dem on und off nicht, da ich davon ausgehe, dass das ebnen nicht so funkltioniert wie in latex, dass der scope einer solchen funktion die geschweiften klammern sind
> 
> Dann kann ich mir aber noch nicht vorstellen wie ein skope entsteht, oder packst du dann alles in den Aufruf des TeMpLaTeS? smile

aber hallo, alles rein da!

naja aber sonst muss man es ja irgendwie machen, wenn man was selbst gebasteltes haben mag für code highlighting mit farben etc.

ansonsten muss man sich mit dem default aus markdown zu frieden geben, wenn das bei dir auch geht...:
```cpp
int main(double a=0)
{ // markdown: cpp
    return -1;
}
```
```python
int main(double a=0)
{ // markdown: python
    return -1;
}
```

**Comment by joshinils** [comment]
noch ein beispiel wie das dann bei benutzung aussehen kann. https://wiki.openstreetmap.org/w/index.php?title=Template:Berlin/Tabbar&action=edit

oder aber das ist ein natives template, was alle unterseiten so einbindet... wie man will
